managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedger.java

```diff
@@ -472,4 +472,9 @@ void asyncSetProperties(Map<String, String> properties, final AsyncCallbacks.Set
      * @param promise
      */
     void trimConsumedLedgersInBackground(CompletableFuture<?> promise);
+
+    /**
+     * Roll current ledger if it is full
+     */
+    void rollCurrentLedgerIfFull();
```

Comment:
Why do we need to expose this in the interface, it should be better to keep in implementation details

Reply:
I hope to be able to consult you here for advice, because I am not sure whether doing type casting in the `BrokerService` is a suitable implementation. Maybe I can modify with:
```java
((ManagedLedgerImpl) managedLedger).rollCurrentLedgerIfFull();
```

Reply:
I was thinking that this is a managedLedger internal task, for which the `BrokerService` shouldn't be concerned. For that it would be better to handle in the `ManagedLedgerFactoryImpl`, to go through all open managed ledger instances and check if a rollover has to be forced.

Reply:
Now I feel a little confused. If we treat rollover as an internal task, should `consumedLedgersMonitor` or `backlogQuotaChecker` be internal tasks as well? Or is my understanding biased?

Reply:
@wuzhanpeng I think you can add an issue to track the monitor that outside the managed ledger but should be maintained by managed ledger own.